Modpack download fails silently when modpackUrl is incorrect

I recently misconfigured my server-setup-config.yaml with an incorrect modpackUrl, specifically for a zip file which didn't exist in forgecdn because I had the correct name, but wrong path to the file.

My server appeared to update and start up correctly:
```
[02:57:08] [INFO] Downloaded the modpack zip file to /home/minecraft/fabric/modpack-download.zip
[02:57:08] [INFO] Moved the mods folder
[02:57:08] [INFO] Starting to unzip files.
[02:57:08] [INFO] Done unzipping the files.
```
But when I tried to connect, my client and server were running different sets of mods. I'm not sure what determined the set of mods loaded, but it did contain mods from the previous versions of the modpack I was using.

Upon further inspection, the contents of the `modpack-download.zip` were garbage, and both the download and extract portions of the startup had silently failed. This was the contents of the `modpack-download.zip` file:
```
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>AccessDenied</Code><Message>Access Denied</Message><RequestId>5WAG9VFQAT1YBGFT</RequestId><HostId>IM64gcBiudURLRRZkkaQHfEfokDfcAvjnuwaXFAtec+0Ulp00lN0q6Y+vKABJp1I7TsDvv3SxNc=</HostId></Error>
```
It appears that the request completed and had a non-empty body, but ultimately that body was an error message.

Instead of this behavior, it should print the error, similar to this case where the hostname can't be resolved:
```
[03:03:07] [INFO] Attempting to download modpack Zip.
[03:03:07] [ERROR] Pack could not be downloaded
[03:03:07] [ERROR] Error while installing pack

java.net.UnknownHostException: media.garbage.net: Name or service not known
        at java.net.Inet6AddressImpl.lookupAllHostAddr(Native Method)
        at java.net.InetAddress$2.lookupAllHostAddr(InetAddress.java:929)
        at java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1324)
        at java.net.InetAddress.getAllByName0(InetAddress.java:1277)
        at java.net.InetAddress.getAllByName(InetAddress.java:1193)
        at java.net.InetAddress.getAllByName(InetAddress.java:1127)
        at okhttp3.Dns$1.lookup(Dns.java:40)
        at okhttp3.internal.connection.RouteSelector.resetNextInetSocketAddress(RouteSelector.java:185)
...
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:200)
        at okhttp3.RealCall.execute(RealCall.java:77)
        at atm.bloodworkxgaming.serverstarter.InternetManager.downloadToFile(InternetManager.kt:51)

```
It appears that the InternetManager does not check the return code from the HTTP request to verify that the request was successful, and swallows the error response as if it was successful:
https://github.com/BloodyMods/ServerStarter/blob/be94470e4b51e06660d9d973af5484632d31d355/src/main/kotlin/atm/bloodworkxgaming/serverstarter/InternetManager.kt#L51-L54

I also think that these errors (both host unreachable and access denied, as well as unzip errors) should stop the starter from continuing, but that's possibly out of scope for this issue. Logging the error properly would be sufficient to help diagnose the misconfiguration.

--- a/src/main/kotlin/atm/bloodworkxgaming/serverstarter/InternetManager.kt
+++ b/src/main/kotlin/atm/bloodworkxgaming/serverstarter/InternetManager.kt
@@ -51,6 +51,7 @@ object InternetManager {
         val res = httpClient.newCall(req).execute()
         val source = res?.body()?.source()
 
+        if (!res.isSuccessful) throw IOException("HTTP error code: ${res.code()} for $url")
         source ?: throw IOException("Message body or source from $url was null")
 
         source.use {

